File: euler3.rb
def factors(n)
	i = Integer.sqrt(n) 
	top = 0
	x = 2
	while x < i
		if n % x == 0
			n = n / x
			if prime(x) == x
				top = x
			end
			x += 1
		else
			x += 1
		end
	end
return top 
end

def prime(n)
	j = Integer.sqrt(n)
	k = 2
	while k < j
		if n % k == 0
			return 0
			break
		else
			k+=1
 	       	end
	end
return n
end

puts factors(600851475143)

File: euler3.c
#include<stdio.h>
#include<math.h>

int is_prime(int n);
int lpf(long long n); //Largest prime factor

int main (){
  int result = 0;
  long long pos_prime = 600851475143;
  result = lpf(pos_prime);
  printf("Largest Prime Factor: %d\n", result); 
  return 0;
}

int is_prime(int n){
  int x = 2;
  while ( x <= sqrt(n) )
    if ( n % x == 0)
      return 0;
    else
      x++;
  return n;
}

int lpf(long long n){
  int x = 2;
  int largest = 0;
  while ( x < sqrt(n) ){ 
    if ( ( is_prime(x) == x ) && ( n % x == 0 ) && ( x > largest ) ){
        largest = x;
    }
    else{
      x++;
    }
  }
  return largest;
}
